### ISSUE TITLE
CSC crashes on invalid LINQ syntax

### ISSUE THREAD

**Issue by sergey-shandar** [dotnet/roslyn#11063]
**Version Used**: 1.2.0.60317 (the version shipped with Visual Studio 2015 Update 2)

**Steps to Reproduce**:

1. Compile this code 
```csharp
using System;

namespace PureLogic
{
    public class Bag
    {
    }

    public static class SelectManyX
    {
        public static Bag Select(this Bag input, Func<int, int> func)
            => null;

        public static Bag A()
            => from r in Bag select 7;
    }
}
```

**Expected Behavior**:
Compilation error.

**Actual Behavior**:
Crash.

**Comment by leppie** [comment]
Probably the same as #10885

**Comment by sergey-shandar** [comment]
It looks similar. The main difference is the Select function instead of deriving from List<...>

**Comment by leppie** [comment]
`Select` is available on `List<T>` via an extension method, so it is still the same.

I was actually trying to test that with a repro like yours (to see the extent) but did not get around to finishing it :+1:

**Comment by gafter** [comment]
This was fixed in #11686.
